frontend/src/main/java/com/google/tripmeout/frontend/servlets/PlaceParentServlet.java

```diff
@@ -0,0 +1,110 @@
+package com.google.tripmeout.frontend.servlets;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.InvalidRequestException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceParentServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+
+  private static final Pattern TRIP_NAME_PATTERN = Pattern.compile(".*/trips/([^/]+)/placeVisits");
+
+  private final GeoApiContext CONTEXT =
+      new GeoApiContext.Builder().apiKey("AIzaSyBbXCXC2uWv3baNmirLtqUYbFsFwCXqLV8").build();
+
+  public PlaceParentServlet(PlaceVisitStorage placeStorage, Gson gson) {
+    this.placeStorage = placeStorage;
+    this.gson = gson;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      Optional<PlaceVisitModel> place =
+          ServletUtil.extractFromRequestBody(request, gson, PlaceVisitModel.class);
+
+      String tripId = ServletUtil.matchUriOrThrowError(request, TRIP_NAME_PATTERN).group(1);
+      logger.atInfo().log("finished matching");
+
+      if (!place.isPresent()) {
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+
+      PlaceDetailsRequest placeRequest =
+          new PlaceDetailsRequest(CONTEXT)
+              .placeId(place.get().placeId())
+              .fields(PlaceDetailsRequest.FieldMask.GEOMETRY_LOCATION,
+                  PlaceDetailsRequest.FieldMask.NAME);
+
+      PlaceDetails details;
+      details = placeRequest.await();
+
+      PlaceVisitModel newPlace = PlaceVisitModel.builder()
+                                     .setTripId(tripId)
+                                     .setPlaceId(place.get().placeId())
+                                     .setUserMark(PlaceVisitModel.UserMark.YES)
+                                     .setName(details.name)
```

Comment:
Much like the other PR, this line confuses me and I clearly don't understand the intent behind your implementation. Don't submit either of these till we talk about it.